tendenci/apps/memberships/views.py

```diff
@@ -2350,7 +2350,32 @@ def report_member_roster(request, template_name='reports/membership_roster.html'
 def report_member_quick_list(request, template_name='reports/membership_quick_list.html'):
     """ Table view of current members fname, lname and company only.
     """
-    members = MembershipDefault.objects.filter(status=1, status_detail="active").order_by('user__last_name')
+    order_by = request.GET.get('order_by', 'user__last_name')
+    
+    descending = order_by.startswith('-')
+    if descending:
+        order_by =  order_by[1:]
+    
+    members = MembershipDefault.objects.filter(status=1, status_detail='active')
+    
+    if order_by == 'member_number_int':
+        members = members.annotate(member_number_int=Cast('member_number', PositiveIntegerField()))
```

Comment:
The `member_number` is a char field. Trying to convert it to an integer wouldn't always work.

Reply:
Good point. Made it optional. Essentially the default template will sort by `member_number` and a local template (override) can implement `member_number_int` for associations that use exclusively int member numbers. I am guessing it was a string to support alphanumeric numbers (akin to car registration plates), but it is a fair assumption that many, like us, have no truck for that and only want to use numbers ... they (including us) can tweak the template locally by adding `_int` to the template tag in the right spot (we will).